lib/mapper.js

```diff
@@ -147,15 +147,19 @@ function common(facade){
   if (facade.city()) ret.city = facade.city();
   if (facade.region()) ret.region = facade.region();
 
+//Older segment clients sent idfa, newer ones send advertisingId
   if (os && os.toString().toLowerCase() == 'ios') {
       ret.idfa = facade.proxy('device.idfa');
-  } else if (os && os.toString().toLowerCase() == 'android') {
-      //Older segment clients sent idfa, newer ones send advertisingId
+      if (!ret.idfa) {
+        ret.idfa = facade.proxy('device.advertisingId');
+      }
+    }
+  else if (os && os.toString().toLowerCase() == 'android') {
```

Comment:
Since the logic in the branches is the same, this might be easier:

```
var advertisingId = facade.proxy('device.advertisingId') || facade.proxy('device.idfa');

if (ios) ret.idfa = advertisingId;
if (android) ret.adid = advertisingId;
```